jax/interpreters/masking.py

```diff
@@ -326,7 +330,7 @@ def _parse_dim(spec):
     return prod(map(_parse_dim, spec.split('*')))
   elif spec.isdigit() or spec.startswith('-') and spec[1:].isdigit():
     return _parse_lit(spec)
-  elif spec in _identifiers:
+  elif spec[0] in _identifiers:
```

Comment:
I am not sure I understand the original intent behind this code. It seems to work for single-letter identifiers. I thought that it makes sense to allow multi-letter identifiers that start with lowercase.